pairing/bn254/suite.go

```diff
@@ -132,9 +132,10 @@ func (s *Suite) Pair(p1 kyber.Point, p2 kyber.Point) kyber.Point {
 	return s.GT().Point().(*pointGT).Pair(p1, p2)
 }
 
+// NB: Not safe for concurrent calls
 func (s *Suite) ValidatePairing(p1, p2, inv1, inv2 kyber.Point) bool {
-	p2.(*pointG2).g.MakeAffine()
-	inv2.(*pointG2).g.MakeAffine()
+	p2.Clone().(*pointG2).g.MakeAffine()
+	inv2.Clone().(*pointG2).g.MakeAffine()
 	return s.Pair(p1, p2).Equal(s.Pair(inv1, inv2))
```

Comment:
I don't think Clone() works like that, it returns a clone of the point, no?

You should be creating local variables if you want to use Clone.

Reply:
🤦‍♂️ indeed you are right. I got mixed up with `MakeAffine` modifying the point itself...